### ISSUE TITLE
DSub (Subsonic Android App) not playing songs

### ISSUE THREAD

**Issue by Larnal** [ampache/ampache#1203]
# Environment
 
 * Ampache version: 3.8.3-develop
 * Web server + version: Nginx 1.6.2
 * Server operating system: Linux osmc 4.4.3-3-osmc (from Raspbian, on Raspbery pi 2)
 * Client operating system: Android 6
 * Client type: 
   - [ ] Web
   - [ ] Ampache
   - [X] SubSonic
   - [ ] Plex
   - [ ] DAAP
   - [ ] UPnP
   - [ ] WebDAV
 * Client application + version: DSub 5.1.6
 
# Settings
I run ampache on Nginx server with a SSL connexion on port 443 (HTTPS)
I use the Nginx config file provided by ampache on the wiki.

In my ampache.cfg.php, the local web path is set to : 
local_web_path = "https://127.0.0.1:443"

# Description
I can't play songs with DSub application on Android. The app connects to the server and I am able to navigate through the music, see the album arts and all, but when I play a song it's stays on 0 B (no download). Caching song doesn't work either.

The Subsonic backend seems to work since another Subsonic app, Subsonic Music Streamer, works very well.

I contacted the DSub developper by reporting this issue on his git repo but he says that might be related to errors in the url : https://github.com/daneren2005/Subsonic/issues/671

Maybe you will be able to help me with that. Thank you very much

# Logs
[Ampache Access logs]
```
89.158.184.108 - - [12/Apr/2016:12:25:46 +0200] "POST /rest/getRandomSongs.view?u=Neuromancer&p=enc:66653038323032323735&v=1.2.0&c=android&size=50 HTTP/1.1" 200 22255 "-" "-"
89.158.184.108 - - [12/Apr/2016:12:25:46 +0200] "POST /rest/getUser.view?u=Neuromancer&p=enc:66653038323032323735&v=1.2.0&c=DSub&username=Neuromancer HTTP/1.1" 200 275 "-" "DSub"
89.158.184.108 - - [12/Apr/2016:12:25:46 +0200] "POST /rest/getBookmarks.view?u=Neuromancer&p=enc:66653038323032323735&v=1.2.0&c=DSub HTTP/1.1" 200 159 "-" "DSub"
89.158.184.108 - - [12/Apr/2016:12:25:46 +0200] "POST /rest/getAvatar.view?u=Neuromancer&p=enc:66653038323032323735&v=1.2.0&c=DSub&username=Neuromancer HTTP/1.1" 200 5 "-" "DSub"
127.0.0.1 - - [12/Apr/2016:12:25:47 +0200] "GET /play/ssid/0ee69dce398f039dbd043a5419ade921/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon%20Tobin%20-%20Dark%20Jovian.$
89.158.184.108 - - [12/Apr/2016:12:25:47 +0200] "POST /rest/stream.view?u=Neuromancer&p=enc:66653038323032323735&v=1.2.0&c=DSub&id=300005005&maxBitRate=320 HTTP/1.1" 404 162 "-" "DSub"
127.0.0.1 - - [12/Apr/2016:12:25:47 +0200] "GET /play/ssid/d61902bd960dce35b7681ef335dfc9de/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon%20Tobin%20-%20Dark%20Jovian.$
89.158.184.108 - - [12/Apr/2016:12:25:47 +0200] "POST /rest/stream.view?u=Neuromancer&p=enc:66653038323032323735&v=1.2.0&c=DSub&id=300005005&maxBitRate=320 HTTP/1.1" 404 162 "-" "DSub"
89.158.184.108 - - [12/Apr/2016:12:25:47 +0200] "POST /rest/getAlbumList.view?u=Neuromancer&p=enc:66653038323032323735&v=1.2.0&c=DSub&type=newest&size=20&offset=0 HTTP/1.1" 200 1395 "-" "DSub"
127.0.0.1 - - [12/Apr/2016:12:25:47 +0200] "GET /play/ssid/6bfd1b9285d94331bc7004430687f119/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon%20Tobin%20-%20Dark%20Jovian.$
89.158.184.108 - - [12/Apr/2016:12:25:47 +0200] "POST /rest/stream.view?u=Neuromancer&p=enc:66653038323032323735&v=1.2.0&c=DSub&id=300005005&maxBitRate=320 HTTP/1.1" 404 162 "-" "DSub"
89.158.184.108 - - [12/Apr/2016:12:25:48 +0200] "POST /rest/getCoverArt.view?u=Neuromancer&p=enc:66653038323032323735&v=1.2.0&c=DSub&id=200000452 HTTP/1.1" 200 97541 "-" "DSub"
127.0.0.1 - - [12/Apr/2016:12:25:48 +0200] "GET /play/ssid/f3e2adad60c541e78904fe9435954335/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon%20Tobin%20-%20Dark%20Jovian.$
89.158.184.108 - - [12/Apr/2016:12:25:48 +0200] "POST /rest/stream.view?u=Neuromancer&p=enc:66653038323032323735&v=1.2.0&c=DSub&id=300005005&maxBitRate=320 HTTP/1.1" 404 162 "-" "DSub"
127.0.0.1 - - [12/Apr/2016:12:25:48 +0200] "GET /play/ssid/2df3dac33c54354a8773a0916d2803cf/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon%20Tobin%20-%20Dark%20Jovian.$
89.158.184.108 - - [12/Apr/2016:12:25:48 +0200] "POST /rest/stream.view?u=Neuromancer&p=enc:66653038323032323735&v=1.2.0&c=DSub&id=300005005&maxBitRate=320 HTTP/1.1" 404 162 "-" "DSub"
127.0.0.1 - - [12/Apr/2016:12:25:48 +0200] "GET /play/ssid/f76b9f07f72240454fa21923e13afb21/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon%20Tobin%20-%20Dark%20Jovian.$
89.158.184.108 - - [12/Apr/2016:12:25:48 +0200] "POST /rest/stream.view?u=Neuromancer&p=enc:66653038323032323735&v=1.2.0&c=DSub&id=300005005&maxBitRate=320 HTTP/1.1" 404 162 "-" "DSub"
127.0.0.1 - - [12/Apr/2016:12:25:48 +0200] "GET /play/ssid/0fab9198eefcba3c95ce59a6547e3f9b/type/song/oid/4999/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon%20Tobin%20-%20In%20Your%20Ow$
89.158.184.108 - - [12/Apr/2016:12:25:48 +0200] "POST /rest/stream.view?u=Neuromancer&p=enc:66653038323032323735&v=1.2.0&c=DSub&id=300004999&maxBitRate=320 HTTP/1.1" 404 162 "-" "DSub"
127.0.0.1 - - [12/Apr/2016:12:25:48 +0200] "GET /play/ssid/95a0b7dce3a2a2688966a8bc3f57b40f/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon%20Tobin%20-%20Dark%20Jovian.$
89.158.184.108 - - [12/Apr/2016:12:25:48 +0200] "POST /rest/stream.view?u=Neuromancer&p=enc:66653038323032323735&v=1.2.0&c=DSub&id=300005005&maxBitRate=320 HTTP/1.1" 404 162 "-" "DSub"
127.0.0.1 - - [12/Apr/2016:12:25:49 +0200] "GET /play/ssid/802fcc0a74fefb82876c9671b2e3a3eb/type/song/oid/4999/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon%20Tobin%20-%20In%20Your%20Ow$
89.158.184.108 - - [12/Apr/2016:12:25:49 +0200] "POST /rest/stream.view?u=Neuromancer&p=enc:66653038323032323735&v=1.2.0&c=DSub&id=300004999&maxBitRate=320 HTTP/1.1" 404 162 "-" "DSub"
127.0.0.1 - - [12/Apr/2016:12:25:49 +0200] "GET /play/ssid/c25c8dda00dccff30e098c401b3a69c2/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon%20Tobin%20-%20Dark%20Jovian.$
89.158.184.108 - - [12/Apr/2016:12:25:49 +0200] "POST /rest/stream.view?u=Neuromancer&p=enc:66653038323032323735&v=1.2.0&c=DSub&id=300005005&maxBitRate=320 HTTP/1.1" 404 162 "-" "DSub"
127.0.0.1 - - [12/Apr/2016:12:25:49 +0200] "GET /play/ssid/8240718f7e5f26f33bf36a1c51af16f4/type/song/oid/4999/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon%20Tobin%20-%20In%20Your%20Ow$
89.158.184.108 - - [12/Apr/2016:12:25:49 +0200] "POST /rest/stream.view?u=Neuromancer&p=enc:66653038323032323735&v=1.2.0&c=DSub&id=300004999&maxBitRate=320 HTTP/1.1" 404 162 "-" "DSub"
89.158.184.108 - - [12/Apr/2016:12:25:49 +0200] "POST /rest/getAlbumList.view?u=Neuromancer&p=enc:66653038323032323735&v=1.2.0&c=DSub&type=highest&size=20&offset=0 HTTP/1.1" 200 160 "-" "DSub"
127.0.0.1 - - [12/Apr/2016:12:25:49 +0200] "GET /play/ssid/63cdf85ae8ee94e40e225181e62568b4/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon%20Tobin%20-%20Dark%20Jovian.$
89.158.184.108 - - [12/Apr/2016:12:25:49 +0200] "POST /rest/stream.view?u=Neuromancer&p=enc:66653038323032323735&v=1.2.0&c=DSub&id=300005005&maxBitRate=320 HTTP/1.1" 404 162 "-" "DSub"
127.0.0.1 - - [12/Apr/2016:12:25:49 +0200] "GET /play/ssid/eb44eeff6e6c2d3d8524a5b9818e770a/type/song/oid/4999/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon%20Tobin%20-%20In%20Your%20Ow$
89.158.184.108 - - [12/Apr/2016:12:25:49 +0200] "POST /rest/stream.view?u=Neuromancer&p=enc:66653038323032323735&v=1.2.0&c=DSub&id=300004999&maxBitRate=320 HTTP/1.1" 404 162 "-" "DSub"
127.0.0.1 - - [12/Apr/2016:12:25:50 +0200] "GET /play/ssid/0a276ca7d7cfd1f879182e2a3d4d521c/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon%20Tobin%20-%20Dark%20Jovian.$
89.158.184.108 - - [12/Apr/2016:12:25:50 +0200] "POST /rest/stream.view?u=Neuromancer&p=enc:66653038323032323735&v=1.2.0&c=DSub&id=300005005&maxBitRate=320 HTTP/1.1" 404 162 "-" "DSub"
```

[Ampache Error log]
```
2016/04/12 12:25:47 [error] 21221#0: *427 open() "/var/www/ampache/play/ssid/0ee69dce398f039dbd043a5419ade921/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon Tobin - Da$
2016/04/12 12:25:47 [error] 21218#0: *430 open() "/var/www/ampache/play/ssid/d61902bd960dce35b7681ef335dfc9de/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon Tobin - Da$
2016/04/12 12:25:47 [error] 21218#0: *432 open() "/var/www/ampache/play/ssid/6bfd1b9285d94331bc7004430687f119/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon Tobin - Da$
2016/04/12 12:25:48 [error] 21218#0: *435 open() "/var/www/ampache/play/ssid/f3e2adad60c541e78904fe9435954335/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon Tobin - Da$
2016/04/12 12:25:48 [error] 21221#0: *437 open() "/var/www/ampache/play/ssid/2df3dac33c54354a8773a0916d2803cf/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon Tobin - Da$
2016/04/12 12:25:48 [error] 21221#0: *439 open() "/var/www/ampache/play/ssid/f76b9f07f72240454fa21923e13afb21/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon Tobin - Da$
2016/04/12 12:25:48 [error] 21221#0: *442 open() "/var/www/ampache/play/ssid/0fab9198eefcba3c95ce59a6547e3f9b/type/song/oid/4999/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon Tobin - In$
2016/04/12 12:25:48 [error] 21218#0: *444 open() "/var/www/ampache/play/ssid/95a0b7dce3a2a2688966a8bc3f57b40f/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon Tobin - Da$
2016/04/12 12:25:49 [error] 21218#0: *446 open() "/var/www/ampache/play/ssid/802fcc0a74fefb82876c9671b2e3a3eb/type/song/oid/4999/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon Tobin - In$
2016/04/12 12:25:49 [error] 21218#0: *448 open() "/var/www/ampache/play/ssid/c25c8dda00dccff30e098c401b3a69c2/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon Tobin - Da$
2016/04/12 12:25:49 [error] 21220#0: *450 open() "/var/www/ampache/play/ssid/8240718f7e5f26f33bf36a1c51af16f4/type/song/oid/4999/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon Tobin - In$
2016/04/12 12:25:49 [error] 21221#0: *453 open() "/var/www/ampache/play/ssid/63cdf85ae8ee94e40e225181e62568b4/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon Tobin - Da$
2016/04/12 12:25:49 [error] 21221#0: *455 open() "/var/www/ampache/play/ssid/eb44eeff6e6c2d3d8524a5b9818e770a/type/song/oid/4999/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon Tobin - In$
2016/04/12 12:25:50 [error] 21221#0: *457 open() "/var/www/ampache/play/ssid/0a276ca7d7cfd1f879182e2a3d4d521c/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon Tobin - Da$
2016/04/12 12:25:50 [error] 21221#0: *459 open() "/var/www/ampache/play/ssid/a71c5688d4e443ceffaab2feace196e2/type/song/oid/4999/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon Tobin - In$
2016/04/12 12:25:50 [error] 21221#0: *461 open() "/var/www/ampache/play/ssid/ca46bf9b083bb6d333b73e876f02b9da/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon Tobin - Da$
2016/04/12 12:25:50 [error] 21221#0: *463 open() "/var/www/ampache/play/ssid/a293cb414099cc784038572bebea6898/type/song/oid/4999/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon Tobin - In$
2016/04/12 12:25:51 [error] 21221#0: *465 open() "/var/www/ampache/play/ssid/3d082b3391aca40e991fe95e7329f0b1/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon Tobin - Da$
2016/04/12 12:25:51 [error] 21221#0: *467 open() "/var/www/ampache/play/ssid/67eae327aeb1dec3a9a76c2a79f245f8/type/song/oid/5004/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon Tobin - Ad$
2016/04/12 12:25:51 [error] 21218#0: *469 open() "/var/www/ampache/play/ssid/a439debd5c177540eee645f830b31c41/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon Tobin - Da$
2016/04/12 12:25:51 [error] 21220#0: *471 open() "/var/www/ampache/play/ssid/328757d3ce4fefe75c71297ea8e8c383/type/song/oid/5004/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon Tobin - Ad$
2016/04/12 12:25:52 [error] 21220#0: *473 open() "/var/www/ampache/play/ssid/94559ae38c4799235534413162394347/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon Tobin - Da$
2016/04/12 12:25:52 [error] 21220#0: *475 open() "/var/www/ampache/play/ssid/2f7547741d7ebcb6d4197adcb297b535/type/song/oid/5004/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Amon Tobin - Ad$
```

[Ampache log]
```
2016-04-12 12:25:46 [Neuromancer] (media) -> Using default target format
2016-04-12 12:25:46 [Neuromancer] (media) -> Transcode settings: from flac to mp3
2016-04-12 12:25:46 [Neuromancer] (media) -> Command: avconv Arguments: -i %FILE% -vn -b:a %BITRATE%K -c:a libmp3lame -f mp3 pipe:1
2016-04-12 12:25:46 [Neuromancer] (media) -> Using default target format
2016-04-12 12:25:46 [Neuromancer] (media) -> Transcode settings: from flac to mp3
2016-04-12 12:25:46 [Neuromancer] (media) -> Command: avconv Arguments: -i %FILE% -vn -b:a %BITRATE%K -c:a libmp3lame -f mp3 pipe:1
2016-04-12 12:25:46 [Neuromancer] (media) -> Using default target format
2016-04-12 12:25:46 [Neuromancer] (media) -> Transcode settings: from flac to mp3
2016-04-12 12:25:46 [Neuromancer] (media) -> Command: avconv Arguments: -i %FILE% -vn -b:a %BITRATE%K -c:a libmp3lame -f mp3 pipe:1
2016-04-12 12:25:46 [Neuromancer] (media) -> Using default target format
2016-04-12 12:25:46 [Neuromancer] (media) -> Transcode settings: from flac to mp3
2016-04-12 12:25:46 [Neuromancer] (media) -> Command: avconv Arguments: -i %FILE% -vn -b:a %BITRATE%K -c:a libmp3lame -f mp3 pipe:1
2016-04-12 12:25:46 [Neuromancer] (media) -> Using default target format
2016-04-12 12:25:46 [Neuromancer] (media) -> Transcode settings: from flac to mp3
2016-04-12 12:25:46 [Neuromancer] (media) -> Command: avconv Arguments: -i %FILE% -vn -b:a %BITRATE%K -c:a libmp3lame -f mp3 pipe:1
2016-04-12 12:25:46 [Neuromancer] (media) -> Using default target format
2016-04-12 12:25:46 [Neuromancer] (media) -> Transcode settings: from flac to mp3
2016-04-12 12:25:46 [Neuromancer] (media) -> Command: avconv Arguments: -i %FILE% -vn -b:a %BITRATE%K -c:a libmp3lame -f mp3 pipe:1
2016-04-12 12:25:46 [Neuromancer] (media) -> Using default target format
2016-04-12 12:25:46 [Neuromancer] (media) -> Transcode settings: from flac to mp3
2016-04-12 12:25:46 [Neuromancer] (media) -> Command: avconv Arguments: -i %FILE% -vn -b:a %BITRATE%K -c:a libmp3lame -f mp3 pipe:1
2016-04-12 12:25:46 [Neuromancer] (media) -> Using default target format
2016-04-12 12:25:46 [Neuromancer] (media) -> Transcode settings: from m4a to mp3
2016-04-12 12:25:46 [Neuromancer] (media) -> Command: avconv Arguments: -i %FILE% -vn -b:a %BITRATE%K -c:a libmp3lame -f mp3 pipe:1
2016-04-12 12:25:47 [Neuromancer] (media) -> Using default target format
2016-04-12 12:25:47 [Neuromancer] (media) -> Transcode settings: from flac to mp3
2016-04-12 12:25:47 [Neuromancer] (media) -> Command: avconv Arguments: -i %FILE% -vn -b:a %BITRATE%K -c:a libmp3lame -f mp3 pipe:1
2016-04-12 12:25:47 [Neuromancer] (media) -> Changing play url type from {flac} to {mp3} due to encoding settings...
2016-04-12 12:25:47 [Neuromancer] (session) -> Session created: 0ee69dce398f039dbd043a5419ade921
2016-04-12 12:25:47 [Neuromancer] (subsonic) -> Stream proxy: https://127.0.0.1:443/play/ssid/0ee69dce398f039dbd043a5419ade921/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/nam$
2016-04-12 12:25:47 [Neuromancer] (media) -> Using default target format
2016-04-12 12:25:47 [Neuromancer] (media) -> Transcode settings: from flac to mp3
2016-04-12 12:25:47 [Neuromancer] (media) -> Command: avconv Arguments: -i %FILE% -vn -b:a %BITRATE%K -c:a libmp3lame -f mp3 pipe:1
2016-04-12 12:25:47 [Neuromancer] (media) -> Changing play url type from {flac} to {mp3} due to encoding settings...
2016-04-12 12:25:47 [Neuromancer] (session) -> Session created: d61902bd960dce35b7681ef335dfc9de
2016-04-12 12:25:47 [Neuromancer] (subsonic) -> Stream proxy: https://127.0.0.1:443/play/ssid/d61902bd960dce35b7681ef335dfc9de/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/nam$
2016-04-12 12:25:47 [Neuromancer] (media) -> Using default target format
2016-04-12 12:25:47 [Neuromancer] (media) -> Transcode settings: from flac to mp3
2016-04-12 12:25:47 [Neuromancer] (media) -> Command: avconv Arguments: -i %FILE% -vn -b:a %BITRATE%K -c:a libmp3lame -f mp3 pipe:1
2016-04-12 12:25:47 [Neuromancer] (media) -> Changing play url type from {flac} to {mp3} due to encoding settings...
2016-04-12 12:25:47 [Neuromancer] (session) -> Session created: 6bfd1b9285d94331bc7004430687f119
2016-04-12 12:25:47 [Neuromancer] (subsonic) -> Stream proxy: https://127.0.0.1:443/play/ssid/6bfd1b9285d94331bc7004430687f119/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/nam$
2016-04-12 12:25:47 [Neuromancer] (media) -> Using default target format
2016-04-12 12:25:47 [Neuromancer] (media) -> Transcode settings: from flac to mp3
2016-04-12 12:25:47 [Neuromancer] (media) -> Command: avconv Arguments: -i %FILE% -vn -b:a %BITRATE%K -c:a libmp3lame -f mp3 pipe:1
2016-04-12 12:25:47 [Neuromancer] (media) -> Changing play url type from {flac} to {mp3} due to encoding settings...
2016-04-12 12:25:47 [Neuromancer] (session) -> Session created: f3e2adad60c541e78904fe9435954335
2016-04-12 12:25:47 [Neuromancer] (subsonic) -> Stream proxy: https://127.0.0.1:443/play/ssid/f3e2adad60c541e78904fe9435954335/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/nam$
```

**Comment by Afterster** [comment]
I don't know why you're getting this behavior, but if other subsonic players are working this is weird.
You shouldn't get the open error on Ampache logs, maybe this is related on how nginx manage url rewriting (open failure, then try rewrite rules... I don't know).
Your logs are truncated so I'm not sure, but maybe it's a transcode issue? DSub could request explitly mp3 transcode whereas other players don't. Could you try with raw mp3 files instead of flac?

**Comment by Larnal** [comment]
Hi, 
Thank you for your help.
I tried with mp3 and got the same issue.
Giving a second look at the error log, it seems the path or name of the song that the app is requesting contains error (this is strange since the "scanning" of the directories see the songs properly). 
This is an untruncated line from the error log
```
2016/04/14 15:40:25 [error] 569#0: *2310 open() "/var/www/ampache/play/ssid/a622575a117bbde6537232b98afb9d99/type/song/oid/4114/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/IO interative - Glacier.mp3" failed (2: No such file or directory), client: 127.0.0.1, server: soundy.cth42.net, request: "GET /play/ssid/a622575a117bbde6537232b98afb9d99/type/song/oid/4114/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/IO%20interative%20-%20Glacier.mp3 HTTP/1.1", host: "127.0.0.1"
```
The name of the file is "01 Glacier.mp3", the tag is for the Artist : "IO Interactive", for the Song name: "Glacier".
Maybe this is due to the naming "Artist - Songname".mp3 ?

By the way, when I open an album, the app seems to launch GET requests for all songs (I see errors for all songs of an album in my error log).

**Comment by lachesis** [comment]
I had a similar error today. I had to explicitly set `http_port = 443` in ampache.cfg.php. I'm not sure if you're hitting the same issue though, since you mentioned having `local_web_path = "https://127.0.0.1:443"`

**Comment by Afterster** [comment]
@Larnal could this be related to https://groups.google.com/forum/#!topic/ampache/h0eXOBzsSTQ?

**Comment by JaylanT** [comment]
I have a similar problem using nginx. Transcoding from flac to opus plays just fine on Dsub, but mp3s remain stuck at downloading 0B. The error log says that there's "No such file or directory" when trying to play mp3s. Other Subsonic clients work just fine. I'm using the nginx config from the wiki.

**Comment by Phyks** [comment]
If I understand correctly, this is a nginx + Ampache + DSub + FLAC to MP3 transcoding issue? (happens only in the following setup)

@JaylanT Could you confirm your workaround with an additional rewrite rule does the trick? Do you have more infos about where it comes from?
If it is indeed the case, I think we can edit the wiki and close this issue.

**Comment by Larnal** [comment]
@JaylanT I added the lines to my nginx config, removed the transcoding for .FLAC files, but I still have the issue.
Now I just use another app on my phone.

**Comment by Phyks** [comment]
Could any of you experiencing this error try with another app as well? Just to confirm it is coming from Ampache (or Ampache + Dsub) and not from Dsub?

Thanks

**Comment by Phyks** [comment]
I was having similar issues with @Niols. We managed to get a working configuration and I can confirm latest version of DSub (as per fdroid repo, v5.0.3) and latest Ampache `develop` branch work just fine behind an nginx, using the config listed in the wiki.

We had to set `local_web_path` equal to `web_path` or the stream proxy was in `127.0.0.1` which the player could not fetch, and gave results very similar to what is described here (and in some related / duplicates issues). This seems to be the case of @Larnal as well, as his log in the first post contains:
```
2016-04-12 12:25:47 [Neuromancer] (subsonic) -> Stream proxy: https://127.0.0.1:443/play/ssid/f3e2adad60c541e78904fe9435954335/type/song/oid/5005/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/nam$
```

I am not sure about the reason why Ampache would send `local_web_path` rather than `web_path`, this is still to be investigated.

**Comment by Phyks** [comment]
Ok, I think I got the root of this issue.

What is happening is that we stream using curl (if available) on the ampache server from the ampache server to the subsonic client. This is done [here](https://github.com/ampache/ampache/blob/develop/lib/class/subsonic_api.class.php#L112). Note that this is highly non intuitive and I guess it is done to handle remote catalogs and local ones the same way. I think we should add a comment there to explain it better.

Then, it will use the `local_web_path` config option, which for most of us is set to its default value, which is `127.0.0.1:80`.

However, most of us have more than one site served by nginx / apache. As the previous URL has no info at all on the host we want, it is handled by the default vhost, which in most cases is not Ampache. Hence, it fails.

Then, subtle part which make this error silent is that we never check it actually works (we don't test the return value from [`curl_exec`](https://github.com/ampache/ampache/blob/develop/lib/class/subsonic_api.class.php#L136). See also [PHP doc](https://secure.php.net/manual/fr/function.curl-error.php). This combined with options and everything makes it hard to find out, and the client is blocked on "loading" as described in this issue.

I will write a PR to handle this and edit the doc accordingly.

**Comment by Niols** [comment]
Infact, the default value for both `web_path` and `local_web_path` is "try to guess from web server config".
I had set `local_web_path` to `127.0.0.1:80` because of the API wiki (Nginx section):

> make sure that locale_web_path is set in your ampache.cfg.php the following way: local_ip:your_http_server_port or at least localhost:your_http_server_port

When we discovered the problem, I set it back to the default ("guess"), and things started working again.

**Comment by Phyks** [comment]
[This PR](https://github.com/ampache/ampache/pull/1289) implements the extra error check I was talking about in my previous comment.

Hopefully, people having similar errors just have the same misconfiguration as we had, since subsonic API with dsub and other clients is working just fine on our instance. I'll let them check it.

@Niols could you edit the wiki according to the setup you used? thanks

**Comment by Larnal** [comment]
@Niols so, do you have an idea of what I should set as value for  `local_web_path` and `web_path`? 
Thank you.

**Comment by Niols** [comment]
I think that you should try setting it to `""` (empty), or to `"your-domain:443"` (not 100% sure though)

**Comment by mourningsun75** [comment]
so for me neither the empty ("") path nor localhost works. :-(

**Comment by mourningsun75** [comment]
I have patched the subsonic_api_class.php with your PR.

Tried:
""
"localhost"
"correct_hostname"

Also with Port and Protocol. Nothing works for me. I cant play or download any song.

**Comment by mourningsun75** [comment]
ok, i have forgotten to switch on debug. so here it is:

```
016-06-24 21:52:15 [user] (session) -> Session created: 672fa8cf8df98c34d9ac41a953fddb0f 
2016-06-24 21:52:15 [user] (subsonic) -> Stream proxy: http://correct_hostname/ampache/play/ssid/672fa8cf8df98c34d9ac41a953fddb0f/type/song/oid/56500/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/-good_music-%20-%20In%20the%20Night.mp3 
2016-06-24 21:52:15 [user] (subsonic) -> Stream error: The requested URL returned error: 404 Not Found 

```

**Comment by Phyks** [comment]
Can you try to wget the stream URL from your server? (If you have shell access)

Else, can you confirm the stream URL works from another computer?

Have you enabled all the correct rewrite rules?

If all the above checks pass, please give extra info on your setup. Thanks

Le 24 juin 2016 21:54:04 UTC+02:00, mourningsun75 <notifications@github.com> a écrit :
>ok, i have forgotten to switch on debug. so here it is:
>
>```
>016-06-24 21:52:15 [user] (session) -> Session created:
>672fa8cf8df98c34d9ac41a953fddb0f 
>2016-06-24 21:52:15 [user] (subsonic) -> Stream proxy:
>http://correct_hostname/ampache/play/ssid/672fa8cf8df98c34d9ac41a953fddb0f/type/song/oid/56500/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/-good_music-%20-%20In%20the%20Night.mp3
>
>2016-06-24 21:52:15 [user] (subsonic) -> Stream error: The requested
>URL returned error: 404 Not Found 
>
>```
>
>---
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/ampache/ampache/issues/1203#issuecomment-228445992

-- 
Envoyé de mon appareil Android avec K-9 Mail. Veuillez excuser ma brièveté.

**Comment by mourningsun75** [comment]
Hi,

before i test again. 
How is the wanted test scenario?
I have installed the actual dev version.
I have not patched anything.

How should i set webpath and local_webpath?

**Comment by Niols** [comment]
I've got a working subsonic configuration with `web_path` and `local_web_path` commented (so that they use the default `""`).

If you are*n't at the root of your webserver*, you will need to set `web_path` to the public path of your Ampache.

If your instance is *behind a proxy*, I think you should set `http_host` and `http_port` to the values that **you** see in your browser.

**Comment by Phyks** [comment]
@mourningsun75 Also note that if you upgrade to latest `develop` branch, after the merge of [this PR](https://github.com/ampache/ampache/pull/1289#event-707320854), you will hopefully have a more verbose debugging in the logs.

**Comment by mourningsun75** [comment]
I have now both (web_path and local_web_path) set to "/ampache". -> cant listen to any song via dsub or download the songs.

when will the pr be merged into develope branch?

**Comment by mourningsun75** [comment]
Just have some time again. Installed the new develope version.

Try to play a song with dsub.
Here is the log:
```
2016-07-08 20:32:16 [admin] (subsonic) -> Stream error: The requested URL returned error: 404 Not Found 
2016-07-08 20:32:17 [admin] (session) -> Session created: 86fc9ed3d10d4ed20d7757c2171edc24 
2016-07-08 20:32:17 [admin] (subsonic) -> Stream proxy: http://server_url:80/ampache/play/ssid/86fc9ed3d10d4ed20d7757c2171edc24/type/song/oid/95169/uid/1/client/DSub/noscrobble/1/bitrate/320/player/api/name/Isobel%20Campbell%20-%20Mark%20Lanegan%20-%20Willow-s%20Song.mp3 
```

after that i tried to download the song with this url and wget. also 404.

**Comment by Phyks** [comment]
So this just means the URL is wrongly generated somehow, if you cannot download the stream proxy. I think you might have a misconfiguration somewhere.

**Comment by mourningsun75** [comment]
But where? :-( i did not change anything. after the last update from the stable version i did not work anymore :-(

**Comment by mourningsun75** [comment]
Hi again... after returned from holidays i installed the latest dev version but nothing changed. :-( Could somebody explain how the .htaccess files have to be written and how the config must look?

**Comment by Phyks** [comment]
I think the doc is up to date on this, @Niols could confirm : https://github.com/ampache/ampache/wiki/API#subsonic-api

We have a working instance with up to date `develop` branch (running begind Nginx), so it should be working.

Can you confirm you followed this wiki page and post some logs when trying to use the subsonic backend?

**Comment by mourningsun75** [comment]
Fixed it with the following config:

web_path = "/ampache"
;local_web_path = "" 

in .htaccess files changed e.g. /play/.... to /ampache/play

**Comment by Phyks** [comment]
This looks like a sensible config. I think the wiki assumes that you are serving files from the root. I will add a note concerning it.

**Comment by Phyks** [comment]
[Done](https://github.com/ampache/ampache/wiki/API#subsonic-api).
